packages/rest/test/unit/rest-server/rest-server.test.ts

```diff
@@ -110,7 +110,9 @@ describe('Application', () => {
   });
 
   function givenRequestContext(rootContext: Context = new Application()) {
-    const server = new RestServer(rootContext, {});
+    const app = rootContext as Application;
```

Comment:
Well, if this helper  requires `rootContext` to be an application, then you should say that in the argument signature.

```ts
function givenRequestContext(app: Application = new Application()) {
  app.server(RestServer);
  // ...
}
```

However, I feel this test is no longer a pure unit-test in such case, since we are bringing too much other dependencies into play.

Is it a necessary requirement that `RestServer` requires an application and that mere root `Context` is not enough?

Reply:
As a rule, unit tests shouldn't be using IoC container (context) to resolve dependencies. Dependencies should be resolved explicitly, using stub classes in most (if not all) cases.